File: perl/97_find_a_writer.pl
open my $fh, '<', $ARGV[0];

for $line ( <$fh> ) {
	$line =~ s/[\n\r]+//g;
	next if $line eq '';

	(my $s, my $raw_args) = split /\|/, $line;
	my @args = split / /, $raw_args;

	shift @args;

	print substr( $line, $_ - 1, 1 ) foreach @args;
	print "\n";
}

close $fh;

exit 0;

File: python/97_find_a_writer.py
import sys

with open(sys.argv[1], 'r') as fh:
    for line in fh.readlines():
        line = line.strip()
        if line == '':
            continue

        (s, coords) = line.split('|')
        coords = [int(c) for c in coords.split(' ') if c != '']

        print ''.join([s[i - 1] for i in coords])

